packages/ui/ToggleItem/ToggleItem.js

```diff
@@ -15,26 +15,49 @@ import kind from '@enact/core/kind';
 import EnactPropTypes from '@enact/core/internal/prop-types';
 import PropTypes from 'prop-types';
 import compose from 'ramda/src/compose';
-import React from 'react';
+import React, {Fragment} from 'react';
 
 import ComponentOverride from '../ComponentOverride';
 import ForwardRef from '../ForwardRef';
+import Slottable from '../Slottable';
 import Toggleable from '../Toggleable';
 import Touchable from '../Touchable';
 
 import componentCss from './ToggleItem.module.less';
 
 // eslint-disable-next-line
-const iconCreator = (position) => ({disabled, icon, iconComponent, selected, iconPosition}) => {
-	if (iconPosition === position) {
+const iconCreator = (position) => ({css, disabled, icon, iconComponent, iconPosition, itemIconAfter, itemIconAfterText, itemIconBefore, itemIconBeforeText, selected}) => {
+	if (position === 'before') {
 		return (
-			<ComponentOverride
-				component={iconComponent}
-				disabled={disabled}
-				selected={selected}
-			>
-				{icon}
-			</ComponentOverride>
+			<Fragment>
+				{itemIconBefore && <ComponentOverride component={itemIconBefore} className={`${css.itemIcon} ${itemIconBefore.props.className}`} />}
```

Comment:
Can we approach to other way to get `className` from applications?

Reply:
I have an idea about how to do this better, but am interested in @ryanjduffy's opinion on this matter.